### PR TITLE
LSRA: Refactor some of the methods around creating kill set

### PR DESCRIPTION
**Pull request by kunalspathak**
I did some refactoring and separated out the methods for the ones that create multiple defs vs. single def as well the one with create defs for call, and the ones that just build the `killMask.` This is in preparation for the upcoming work I am doing to add support for predicate registers. This patch is extracted out from https://github.com/dotnet/runtime/pull/98258

### CODE REVIEW

**Comment by kunalspathak** [comment]
[TP Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=682009&view=ms.vss-build-web.run-extensions-tab)

image
image

**Comment by kunalspathak** [comment]
@dotnet/jit-contrib

**Review by jakobbotsch — APPROVED** [review]

**Comment by kunalspathak** [comment]
@jakobbotsch - i fixed some riscv64 build errors. PTAL

File: src/coreclr/jit/targetriscv64.cpp [hunk 1]
```diff
@@ -131,8 +131,8 @@ ABIPassingInformation RiscV64Classifier::Classify(Compiler*    comp,
             regNumber secondReg = (isSecondFloat ? m_floatRegs : m_intRegs).Dequeue();
 
             return {2, new (comp, CMK_ABI)
-                           ABIPassingSegment[]{ABIPassingSegment::InRegister(firstReg, 0, firstSize),
-                                               ABIPassingSegment::InRegister(secondReg, offset, secondSize)}};
+                           ABIPassingSegment[2]{ABIPassingSegment::InRegister(firstReg, 0, firstSize),
```
Comment (jakobbotsch) [inline src/coreclr/jit/targetriscv64.cpp:134]: Why this change?

Comment (kunalspathak) [reply src/coreclr/jit/targetriscv64.cpp:134]: i get build error locally without it:

```
  [ 68%] Building CXX object jit/CMakeFiles/clrjit_unix_riscv64_x64.dir/targetriscv64.cpp.o
  /home/kpathak/git/runtime/src/coreclr/jit/targetriscv64.cpp: In member function ‘ABIPassingInformation RiscV64Classifier::Classify(Compiler*, var_types, ClassLayout*, WellKnownArg)’:
  /home/kpathak/git/runtime/src/coreclr/jit/targetriscv64.cpp:165:46: error: expected primary-expression before ‘]’ token
    165 |                            ABIPassingSegment[]{passSlot(0, TARGET_POINTER_SIZE),
        |                                              ^
  /home/kpathak/git/runtime/src/coreclr/jit/targetriscv64.cpp:166:111: error: too many initializers for ‘ABIPassingSegment [1]’
    166 |                                                passSlot(TARGET_POINTER_SIZE, passedSize - TARGET_POINTER_SIZE)}};
        |                                                                                                               ^
  make[3]: *** [jit/CMakeFiles/clrjit_unix_riscv64_x64.dir/build.make:1613: jit/CMakeFiles/clrjit_unix_riscv64_x64.dir/targetriscv64.cpp.o] Error 1
  make[2]: *** [CMakeFiles/Makefile2:2685: jit/CMakeFiles/clrjit_unix_riscv64_x64.dir/all] Error 2
  make[1]: *** [CMakeFiles/Makefile2:1681: CMakeFiles/crosscomponents.dir/rule] Error 2
  make: *** [Makefile:299: crosscomponents] Error 2
```

**Comment by kunalspathak** [comment]
/ba-g failure is related to https://github.com/dotnet/runtime/issues/102479
